src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -284,15 +348,20 @@ private WriteConcern GetTransactionWriteConcern()
                 WriteConcern.WMajority;
         }
 
-        private bool ShouldIgnoreAbortTransactionException(Exception ex)
+        private bool ShouldIgnoreAbortTransactionException(Exception exception)
         {
-            var commandException = ex as MongoCommandException;
+            var commandException = exception as MongoCommandException;
             if (commandException != null)
             {
-                return commandException.CodeName == "NoSuchTransaction";
+                return true;
             }
 
             return false;
```

Comment:
So if abort command throws, say, a SocketException, the exception is thrown.  The spec currently says:

    If the second abortTransaciton attempt also fails with a retryable error drivers MUST ignore the error

Let's implement the spec as currently written and then adjust when the spec changes.

Reply:
Good catch. The code as written is not ignoring retryable exceptions on the second attempt.

The spec is ambiguous about what to do on the second attempt with exceptions that don't fall into either the ignorable or retryable category. What do you want to do about this ambiguous case?